pupil_src/shared_modules/video_export/plugins/world_video_exporter.py

```diff
@@ -0,0 +1,292 @@
+    rec_dir,
+    user_dir,
+    min_data_confidence,
+    start_frame,
+    end_frame,
+    plugin_initializers,
+    out_file_path,
+    pre_computed_eye_data,
+):
+    """
+    Simulates the generation for the world video and saves a certain time range as a video.
+    It simulates a whole g_pool such that all plugins run as normal.
+    """
+    from glob import glob
+    from time import time
+
+    import file_methods as fm
+    import player_methods as pm
+    from av_writer import AV_Writer
+
+    # we are not importing manual gaze correction. In Player corrections have already been applied.
+    # in batch exporter this plugin makes little sense.
+    from fixation_detector import Offline_Fixation_Detector
+
+    # Plug-ins
+    from plugin import Plugin_List, import_runtime_plugins
+    from video_capture import EndofVideoError, init_playback_source
+    from vis_circle import Vis_Circle
+    from vis_cross import Vis_Cross
+    from vis_eye_video_overlay import Vis_Eye_Video_Overlay
+    from vis_light_points import Vis_Light_Points
+    from vis_polyline import Vis_Polyline
+    from vis_scan_path import Vis_Scan_Path
+    from vis_watermark import Vis_Watermark
+
+    PID = str(os.getpid())
+    logger = logging.getLogger(__name__ + " with pid: " + PID)
+    start_status = "Starting video export with pid: {}".format(PID)
+    print(start_status)
+    yield start_status, 0
+
+    try:
+        vis_plugins = sorted(
+            [
+                Vis_Circle,
+                Vis_Cross,
+                Vis_Polyline,
+                Vis_Light_Points,
+                Vis_Watermark,
+                Vis_Scan_Path,
+                Vis_Eye_Video_Overlay,
+            ],
+            key=lambda x: x.__name__,
+        )
+        analysis_plugins = [Offline_Fixation_Detector]
+        user_plugins = sorted(
+            import_runtime_plugins(os.path.join(user_dir, "plugins")),
+            key=lambda x: x.__name__,
+        )
+
+        available_plugins = vis_plugins + analysis_plugins + user_plugins
+        name_by_index = [p.__name__ for p in available_plugins]
+        plugin_by_name = dict(zip(name_by_index, available_plugins))
+
+        pm.update_recording_to_recent(rec_dir)
+
+        audio_path = os.path.join(rec_dir, "audio.mp4")
+
+        meta_info = pm.load_meta_info(rec_dir)
+
+        g_pool = GlobalContainer()
+        g_pool.app = "exporter"
+        g_pool.min_data_confidence = min_data_confidence
+
+        valid_ext = (".mp4", ".mkv", ".avi", ".h264", ".mjpeg", ".fake")
+        video_path = [
+            f
+            for f in glob(os.path.join(rec_dir, "world.*"))
+            if os.path.splitext(f)[1] in valid_ext
+        ][0]
+        cap = init_playback_source(g_pool, source_path=video_path, timing=None)
+
+        timestamps = cap.timestamps
+
+        # Out file path verification, we do this before but if one uses a separate tool, this will kick in.
+        if out_file_path is None:
```

Comment:
This should not happen any more, correct?

Reply:
I don't know, we can test it without.